### DISCUSSION TITLE
Adding DAP support for languages

### DISCUSSION THREAD

**Discussion by lukepighetti** [helix-editor/helix#4182] [Q&A]
I am trying to contribute DAP integration for Dart, but I am having a very difficult time understanding the integration and error messages. I understand that this feature is experimental, but I'd still like to experiment with it in my language of choice.

A couple pain points: 

- there are enum configurations that are typed as a string making it harder to understand (like `DebugAdapterConfig.transport`, `tcp|stdio`)
- there don't appear to be any way to populate inputs, like current file name, or current working directory
- general lack of documentation to ease new contributors into adding debug support for languages

Here's the Neovim POC for Dart DAP support https://gist.github.com/christopherfujino/80be0f4cd88f75c4991b478e6b071153#dart-proof-of-concept

I'll probably have more thoughts as I go, but I figured I'd get the ball rolling

**Comment by noor-tg** [comment]
did you figure out how to integrate dart dap with helix?
